### PR TITLE
common: add gosec to golangci linter

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
- Also fix the linter errors
  - Skip G112 for profiler and metrics

### CODE REVIEW

**Review by chrisho — APPROVED** [review]
LGTM, execute `make ci` and the `golangci-lint` does not display any errors, thanks.

**Review by bk201 — APPROVED** [review]
LGTM

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -25,7 +25,8 @@ func Run() {
 	prometheus.MustRegister(KsmdUtilizationGV)
 
 	http.Handle(MetricPath, promhttp.Handler())
-	if err := http.ListenAndServe(MetricPort, nil); err != nil {
+	metricServer := &http.Server{Addr: MetricPort} // #nosec G112
+	if err := metricServer.ListenAndServe(); err != nil {
```
Comment (bk201) [inline pkg/metrics/metrics.go:29]: Please help check if we can add a timeout: https://app.deepsource.com/directory/analyzers/go/issues/GO-S2112

**Comment by bk201** [comment]
Confirm metric server works:
```
harvester-node-manager-wg7vh:/ # curl 10.52.1.58:9526/metrics
# HELP go_cgo_go_to_c_calls_calls_total Count of calls made from Go to C by the current process.
# TYPE go_cgo_go_to_c_calls_calls_total counter
go_cgo_go_to_c_calls_calls_total 0
# HELP go_gc_cycles_automatic_gc_cycles_total Count of completed GC cycles generated by the Go runtime.
# TYPE go_gc_cycles_automatic_gc_cycles_total counter
go_gc_cycles_automatic_gc_cycles_total 10
# HELP go_gc_cycles_forced_gc_cycles_total Count of completed GC cycles forced by the application.
# TYPE go_gc_cycles_forced_gc_cycles_total counter
go_gc_cycles_forced_gc_cycles_total 0
...
```

File: .golangci.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  gosec:
+    # https://github.com/securego/gosec#available-rules
+    excludes:
+      - G101 # Look for hard coded credentials
+      - G108 # Profiling endpoint is automatically exposed on /debug/pprof
+      - G401 # Use of weak cryptographic primitive
+      - G402 # TLS InsecureSkipVerify set true
+      - G505 # Blocklisted import crypto/md5: weak cryptographic primitive
+    config:
+      G306: "0644" # Poor file permissions used when writing to a new file
```
Comment (guangbochen) [inline .golangci.yaml:30]: Do these configurations are all required for this repository? can we enable only if it needed, thanks.

Comment (Vicente-Cheng) [reply .golangci.yaml]: Sure, remove the unused exclusion.
Thanks for the reminder!

File: .golangci.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    - generated
+  deadline: 5m
+  tests: true
+  build-tags:
+    - test
+linters-settings:
+  gosec:
+    # https://github.com/securego/gosec#available-rules
+    config:
+      G306: "0644" # Poor file permissions used when writing to a new file
```
Comment (guangbochen) [inline .golangci.yaml:24]: do we need poor file permission for this repo?

Comment (Vicente-Cheng) [reply .golangci.yaml:24]: Hi @guangbochen, yes, we need that due to some files created with 0644 permission (the default G306 is 0600).

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**